src/vl-map-wmts-layer.js

```diff
@@ -0,0 +1,69 @@
+import {VlMapLayer} from '/src/vl-map-layer.js';
+import {OlWMTSSource, OlWMTSTileGrid, OlTileLayer, OlExtent} from '/node_modules/vl-mapactions/dist/vl-mapactions.js';
+/**
+ * VlMapWmtsLayer
+ *
+ * @class
+ * @classdesc Een WMTS (overlay) layer
+ *
+ * @extends VlMapBaseLayer
+ *
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-map-wmts-layer.html|Demo}
+ */
+export class VlMapWmtsLayer extends VlMapLayer {
+  constructor() {
+    super();
+    this._layer = this._createWmtsLayer();
+  }
+
+  async connectedCallback() {
+    await super.connectedCallback();
+  }
+
+
+  get _projection() {
+    if (this.parentNode) {
+      return this.parentNode._projection;
+    }
+  }
+
+  get url() {
+    return this.getAttribute('data-vl-url');
+  }
+
+  _createWmtsLayer() {
+    const size = OlExtent.getWidth(this._projection.getExtent()) / 256;
+    const resolutions = new Array(16);
+    const matrixIds = new Array(16);
+    for (let z = 0; z < 16; ++z) {
+      resolutions[z] = size / Math.pow(2, z);
+      matrixIds[z] = z;
+    }
+
+    const source = new OlWMTSSource({
+      url: this.url,
+      layer: this.getAttribute('data-vl-layer'),
+      matrixSet: 'BPL72VL',
+      format: 'image/png',
+      projection: this._projection,
+      tileGrid: new OlWMTSTileGrid({
+        extent: this._projection.getExtent(),
+        origin: OlExtent.getTopLeft(this._projection.getExtent()),
+        resolutions: resolutions,
+        matrixIds: matrixIds,
+      }),
+      style: '',
+    });
+
+    const layer = new OlTileLayer({
```

Comment:
In voortoets zit het volgende:

```
		var options = {
			url: config.url,
			layer: config.layer,
			name: config.name,
			gutter: 0,
			buffer: 0,
			isBaseLayer: config.isBaseLayer || false,
			visibility: config.visible || false,
			transitionEffect : config.isBaseLayer ? 'resize': '',
			tileOrigin: new OpenLayers.LonLat(9928, 329072),
			units: "m",
			maxExtent: gisconfig.bounds,
			projection: gisconfig.projection,
			style: '',
			format: "image/png",
			matrixSet: 'BPL72VL'
		};
```

Hier en daar komen enkele dingen niet terug (gutter en buffer), maar is misschien niet meer nodig in OL 6?

Reply:
gutter heb ik nog gevonden bij TileWMS maar stond default op 0, dus leek me niet nuttig om mee te nemen

Reply:
Ok, ik zie ook bv `units` en een hardcoded `tileOrigin` (itt. `OlExtent.getTopLeft(this._projection.getExtent())`) staan?

Reply:
Ah dit is WMTS zie ik net, nee, daarvoor hadden we afgesproken dat we voorlopig enkel voor GRB gingen hardcoderen

Reply:
Ik heb dat overgenomen van hoe WMTS gebruikt werd bij de baselayers @SanderKleykens, en leek te werken alvast ...

Reply:
Ja cava, dan ist goed :)